_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Plugin/AssemblyBuild/CompactDynamicArticleListBuild.php

```diff
@@ -4,27 +4,31 @@
 
 use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
 use Drupal\Core\Entity\EntityInterface;
-use Drupal\assembly\Plugin\AssemblyBuildBase;
-use Drupal\assembly\Plugin\AssemblyBuildInterface;
-use Drupal\node\Entity\Node;
-use Drupal\Core\HttpClient;
 
 /**
- * Displays a list of recent content from Wordpress and Drupal with Disqus comments and related topics
- *  @AssemblyBuild(
+ * Displays a list of recent content from Wordpress and Drupal in a list.
+ *
+ * @AssemblyBuild(
  *   id = "compact_dynamic_article_list",
  *   types = { "compact_dynamic_article_list" },
  *   label = @Translation("Compact Dynamic Article List")
  * )
  */
 class CompactDynamicArticleListBuild extends DynamicContentFeedBuild {
 
+  /**
+   * Build the CompactDynamicArticleList render array.
+   */
   public function build(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
-    $count = 8;
+    $count = $entity->get('field_number_of_posts')->getValue();
+    $count = reset($count)['value'] ?? 8;
```

Comment:
Minor feedback:   It would be helpful to add field helper text on the values allowed (minimum: 4, maximum: 10),
and also if we have a default value of 8, then the field should not become required.

I was able to update a rich text assembly on a page and save, without having to update the compact dynamic article list to provide the missing field, however.